# This notebook is to explore the census track data in geopandas

In [14]:
import geoplot as gplt
import geopandas as gpd
import geoplot.crs as gcrs
import imageio
import folium
import requests
import pathlib

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import mapclassify as mc

%matplotlib inline
pd.set_option('display.max_colwidth', 999)

In [15]:
census_track = gpd.read_file('./data/census_track_2010_cook_county/tl_2010_17031_tract10.shp')
census_track.head()

STATEFP10 COUNTYFP10 TRACTCE10      GEOID10 NAME10         NAMELSAD10  \
0        17        031    840300  17031840300   8403  Census Tract 8403   
1        17        031    840200  17031840200   8402  Census Tract 8402   
2        17        031    841100  17031841100   8411  Census Tract 8411   
3        17        031    841200  17031841200   8412  Census Tract 8412   
4        17        031    838200  17031838200   8382  Census Tract 8382   

  MTFCC10 FUNCSTAT10  ALAND10  AWATER10   INTPTLAT10    INTPTLON10  \
0   G5020          S   831224         0  +41.8320943  -087.6818822   
1   G5020          S   831087         0  +41.8445748  -087.6491915   
2   G5020          S  1143247         0  +41.8510058  -087.6350978   
3   G5020          S   625405         0  +41.8555618  -087.6833420   
4   G5020          S  1157564         0  +41.8704157  -087.6750794   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 geometry  
0  POLYGON ((-87.68608 41.82296, -87.68607 41.82302, -87.68608 41.82331, -87.68608 41.82338, -87.68612 41.82416, -87.68612 41.82419, -87.68613 41.82431, -87.68614 41.82435, -87.68620 41.82442, -87.68628 41.82450, -87.68636 41.82458, -87.68667 41.82489, -87.68663 41.82491, -87.68653 41.82497, -87.68650 41.82499, -87.68647 41.82501, -87.68644 41.82503, -87.68632 41.82510, -87.68625 41.82514, -87.68622 41.82516, -87.68619 41.82518, -87.68620 41.82525, -87.68621 41.82534, -87.68621 41.82547, -87.68621 41.82554, -87.68639 41.82559, -87.68644 41.82566, -87.68656 41.82583, -87.68668 41.82610, -87.68681 41.82635, -87.68682 41.82639, -87.68689 41.82651, -87.68694 41.82663, -87.68703 41.82683, -87.68704 41.82684, -87.68714 41.82709, -87.68724 41.82743, -87.68725 41.82746, -87.68729 41.82767, -87.68729 41.82782, -87.68730 41.82826, -87.68731 41.82841, -87.68731 41.82877, -87.68734 41.82987, -87.68735 41.83024, -87.68735 41.83042, -87.68735 41.83096, -87.68735 41.83115, -87.68736 41.83132, -87.6...  
1  POLYGON ((-87.64633 41.84009, -87.64632 41.83989, -87.64632 41.83986, -87.64631 41.83950, -87.64629 41.83841, -87.64628 41.83805, -87.64649 41.83804, -87.64693 41.83804, -87.64713 41.83804, -87.64734 41.83803, -87.64749 41.83803, -87.64771 41.83802, -87.64795 41.83802, -87.64810 41.83802, -87.64822 41.83802, -87.64858 41.83801, -87.64870 41.83801, -87.64879 41.83801, -87.64884 41.83801, -87.64927 41.83801, -87.64941 41.83800, -87.64983 41.83849, -87.65092 41.83976, -87.65109 41.83996, -87.65152 41.84046, -87.65169 41.84065, -87.65220 41.84125, -87.65237 41.84145, -87.65272 41.84186, -87.65377 41.84308, -87.65384 41.84316, -87.65413 41.84349, -87.65421 41.84360, -87.65443 41.84385, -87.65448 41.84390, -87.65449 41.84392, -87.65455 41.84399, -87.65456 41.84400, -87.65463 41.84408, -87.65463 41.84408, -87.65469 41.84415, -87.65484 41.84433, -87.65486 41.84434, -87.65492 41.84441, -87.65496 41.84446, -87.65507 41.84460, -87.65511 41.84464, -87.65569 41.84531, -87.65588 41.84554, -87.6...  
2  POLYGON ((-87.62935 41.85280, -87.62934 41.85258, -87.62932 41.85200, -87.62931 41.85145, -87.62930 41.85107, -87.6

In [16]:
census_track.shape

(1319, 13)

In [17]:
type(census_track['GEOID10'][0])

str

In [18]:
census_track['GEOID10'].isna().sum()

0

In [19]:
food_access = pd.read_csv('./data/food_access/FoodAccessResearchAtlasData2019.csv')
food_access.head()

CensusTract    State          County  Urban  Pop2010  OHU2010  \
0   1001020100  Alabama  Autauga County      1     1912      693   
1   1001020200  Alabama  Autauga County      1     2170      743   
2   1001020300  Alabama  Autauga County      1     3373     1256   
3   1001020400  Alabama  Autauga County      1     4386     1722   
4   1001020500  Alabama  Autauga County      1    10766     4082   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
0                  0       0.0      0.00                  0  ...   
1                  0     181.0      8.34                  1  ...   
2                  0       0.0      0.00                  0  ...   
3                  0       0.0      0.00                  0  ...   
4                  0     181.0      1.68                  0  ...   

   TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  TractAIAN  \
0         221.0      1622.0       217.0        14.0         0.0       14.0   
1         214.0       888.0      1217.0         5.0         0.0        5.0   
2         439.0      2576.0       647.0        17.0         5.0       11.0   
3         904.0      4086.0       193.0        18.0         4.0       11.0   
4        1126.0      8666.0      1437.0       296.0         9.0       48.0   

   TractOMultir  TractHispanic  TractHUNV  TractSNAP  
0          45.0           44.0        6.0      102.0  
1          55.0           75.0       89.0      156.0  
2         117.0           87.0       99.0      172.0  
3          74.0           85.0       21.0       98.0  
4         310.0          355.0      230.0      339.0  

[5 rows x 147 columns]

In [20]:
type(food_access['CensusTract'][0])

numpy.int64

In [21]:
food_access['CensusTract'].isna().sum()

0

In [22]:
food_access['CensusTract'][0]

1001020100

In [23]:
# convert food_access['CensusTract'] to a string
food_access['CensusTract']= [str(ct_num) for ct_num in food_access['CensusTract']]

In [24]:
type(food_access['CensusTract'][0])

str

In [25]:
food_access['CensusTract'][0]

'1001020100'

In [26]:
food_access.head()

CensusTract    State          County  Urban  Pop2010  OHU2010  \
0  1001020100  Alabama  Autauga County      1     1912      693   
1  1001020200  Alabama  Autauga County      1     2170      743   
2  1001020300  Alabama  Autauga County      1     3373     1256   
3  1001020400  Alabama  Autauga County      1     4386     1722   
4  1001020500  Alabama  Autauga County      1    10766     4082   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...  \
0                  0       0.0      0.00                  0  ...   
1                  0     181.0      8.34                  1  ...   
2                  0       0.0      0.00                  0  ...   
3                  0       0.0      0.00                  0  ...   
4                  0     181.0      1.68                  0  ...   

   TractSeniors  TractWhite  TractBlack  TractAsian  TractNHOPI  TractAIAN  \
0         221.0      1622.0       217.0        14.0         0.0       14.0   
1         214.0       888.0      1217.0         5.0         0.0        5.0   
2         439.0      2576.0       647.0        17.0         5.0       11.0   
3         904.0      4086.0       193.0        18.0         4.0       11.0   
4        1126.0      8666.0      1437.0       296.0         9.0       48.0   

   TractOMultir  TractHispanic  TractHUNV  TractSNAP  
0          45.0           44.0        6.0      102.0  
1          55.0           75.0       89.0      156.0  
2         117.0           87.0       99.0      172.0  
3          74.0           85.0       21.0       98.0  
4         310.0          355.0      230.0      339.0  

[5 rows x 147 columns]

In [27]:
food_access[food_access['County'] == 'Cook County'].shape

(1320, 147)

### Merge census_tract and food_access on CensusTract and GEOID10 including only the data that exist in both files.

In [28]:

food_data = food_access.merge(census_track, left_on = 'CensusTract', right_on = 'GEOID10', how = 'inner')
food_data.head()

CensusTract     State       County  Urban  Pop2010  OHU2010  \
0  17031010100  Illinois  Cook County      1     4854     2302   
1  17031010201  Illinois  Cook County      1     6450     2463   
2  17031010202  Illinois  Cook County      1     2818     1115   
3  17031010300  Illinois  Cook County      1     6236     2826   
4  17031010400  Illinois  Cook County      1     5042     2098   

   GroupQuartersFlag  NUMGQTRS  PCTGQTRS  LILATracts_1And10  ...      GEOID10  \
0                  0     218.0      4.49                  0  ...  17031010100   
1                  0     163.0      2.53                  0  ...  17031010201   
2                  0     315.0     11.18                  0  ...  17031010202   
3                  0     791.0     12.68                  0  ...  17031010300   
4                  0    1349.0     26.76                  0  ...  17031010400   

   NAME10           NAMELSAD10  MTFCC10  FUNCSTAT10  ALAND10  AWATER10  \
0     101     Census Tract 101    G5020           S   381061         0   
1  102.01  Census Tract 102.01    G5020           S   504470         0   
2  102.02  Census Tract 102.02    G5020           S   351716         0   
3     103     Census Tract 103    G5020           S   472017         0   
4     104     Census Tract 104    G5020           S   548754    602605   

    INTPTLAT10    INTPTLON10  \
0  +42.0212617  -087.6698442   
1  +42.0160076  -087.6801485   
2  +42.0160503  -087.6733216   
3  +42.0159407  -087.6665351   
4  +42.0064028  -087.6571350   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 geometry  
0  POLYGON ((-87.66368 42.01940, -87.66384 42.01934, -87.66384 42.01929, -87.66389 42.01929, -87.66402 42.01929, -87.66443 42.01928, -87.66456 42.01927, -87.66486 42.01927, -87.66527 42.01926, -87.66574 42.01925, -87.66604 42.01925, -87.66648 42.01924, -87.66681 42.01924, -87.66780 42.01922, -87.66824 42.01921, -87.66824 42.01924, -87.66824 42.01933, -87.66824 42.01937, -87.66849 42.01936, -87.66887 42.01936, -87.66925 42.01936, -87.66950 42.01936, -87.66963 42.01937, -87.66975 42.01937, -87.67012 42.01937, -87.67049 42.01937, -87.67074 42.01937, -87.67086 42.01937, -87.67121 42.01937, -87.67133 42.01937, -87.67145 42.01937, -87.67180 42.01938, -87.67192 42.01938, -87.67215 42.01938, -87.67251 42.01938, -87.67287 42.01938, -87.67311 42.01939, -87.67336 42.01937, -87.67355 42.01964, -87.67406 42.02006, -87.67634 42.02193, -87.67710 42.02255, -87.67712 42.02263, -87.67718 42.02286, -87.67720 42.02294, -87.67628 42.02296, -87.67597 42.02297, -87.67351 42.02304, -87.67259 42.02306, -87.6...  
1  POLYGON ((-87.68009 42.01254, -87.68027 42.01254, -87.68075 42.01253, -87.68220 42.01250, -87.68234 42.01250, -87.68268 42.01250, -87.68275 42.01246, -87.68276 42.01246, -87.68300 42.01236, -87.68309 42.01232, -87.68313 42.01245, -87.68320 42.01267, -87.68357 42.01369, -87.68369 42.01403, -87.68377 42.01426, -87.68400 42.01493, -87.68408 42.01515, -87.68413 42.01532, -87.68416 42.01541, -87.68417 42.01546, -87.68424 42.01601, -87.68426 42.01622, -87.68430 42.01650, 

In [29]:
food_data.shape

(1314, 160)

#### census tract has 1319 rows, food_access has 1320 rows. When merged food_data has 1314 rows.

In [30]:
food_data.isna().sum().sort_values(ascending = False)

lasnap20share    1314
laasian10        1314
lasnap10         1314
lahunv10share    1314
lahisp10share    1314
                 ... 
LATracts_half       0
LATracts1           0
LATracts10          0
LATracts20          0
geometry            0
Length: 160, dtype: int64

In [31]:
food_data.describe().T

count          mean           std     min       25%  \
Urban              1314.0  9.984779e-01  3.899886e-02     0.0       1.0   
Pop2010            1314.0  3.953330e+03  1.788800e+03   237.0    2548.0   
OHU2010            1314.0  1.496466e+03  7.610063e+02   100.0     964.5   
GroupQuartersFlag  1314.0  3.805175e-03  6.159207e-02     0.0       0.0   
NUMGQTRS           1314.0  6.870776e+01  3.525771e+02     0.0       0.0   
...                   ...           ...           ...     ...       ...   
TractHispanic      1314.0  9.473075e+02  1.265664e+03     1.0     135.0   
TractHUNV          1314.0  2.644140e+02  3.103921e+02     0.0      91.0   
TractSNAP          1314.0  2.258455e+02  1.915820e+02     0.0      77.0   
ALAND10            1314.0  1.845780e+06  3.416277e+06  8299.0  402537.5   
AWATER10           1314.0  5.207878e+04  2.773929e+05     0.0       0.0   

                        50%         75%         max  
Urban                   1.0        1.00         1.0  
Pop2010              3848.0     5149.50     16735.0  
OHU2010              1424.0     1893.00      9751.0  
GroupQuartersFlag       0.0        0.00         1.0  
NUMGQTRS                2.0       30.75     10273.0  
...                     ...         ...         ...  
TractHispanic         397.5     1181.50      6699.0  
TractHUNV             174.5      308.50      3473.0  
TractSNAP             185.0      315.00      1151.0  
ALAND10            869749.0  2072697.50  76741783.0  
AWATER10                0.0        0.00   3802983.0  

[146 rows x 8 columns]

In [32]:
# drop the geometry column for now and then write data to csv
food_data.drop(columns = 'geometry').to_csv('./data/food_data.csv')